adapters/mock-da/src/service.rs

```diff
@@ -79,6 +79,62 @@ impl MockDaService {
         }
         anyhow::bail!("No blob at height={height} has been sent in time")
     }
+
+    async fn add_blob(&self, blob: &[u8], zkp_proof: Vec<u8>) -> Result<u64, anyhow::Error> {
+        let mut blocks = self.blocks.write().await;
+
+        let (previous_block_hash, height) = match blocks.iter().last().map(|b| b.header().clone()) {
+            None => (MockHash::from([0; 32]), 0),
+            Some(block_header) => (block_header.hash(), block_header.height + 1),
+        };
+        let data_hash = hash_to_array(blob);
+        let proof_hash = hash_to_array(&zkp_proof);
+        // Hash only from single blob
+        let block_hash = block_hash(height, data_hash, proof_hash, previous_block_hash.into());
+
+        let blob = MockBlob::new_with_zkp_proof(
+            blob.to_vec(),
+            zkp_proof,
+            self.sequencer_da_address,
+            data_hash,
+        );
+        let header = MockBlockHeader {
+            prev_hash: previous_block_hash,
+            hash: block_hash,
+            height,
+            time: Time::now(),
```

Comment:
Dunno if we want to worry about that on the mock implementation, but using the local systime might be unreliable in terms of consensus. The topic is further discussed [here](https://ethresear.ch/t/time-as-a-public-service-in-byzantine-context/6895)

Reply:
This was changed yesterday: https://github.com/Sovereign-Labs/sovereign-sdk/commit/6e22b4c91bce64b70cb1624770ce10a773c36a08 . We use the `MockDa` only for tests and in the demo (it is always a single node)